include/deal.II/grid/grid_refinement.h

```diff
@@ -300,6 +301,65 @@ namespace GridRefinement
                                const VectorType            &criteria,
                                const unsigned int          order=2);
 
+
+  /**
+   * Return the highest ratio among the dimensions of a @p cell. Moreorer,
+   * return the dimension relative to the highest elongation.
+   *
+   * @param[in] cell : an iterator pointing to the cell.
+   *
+   * @return  a std::pair<unsigned int, double> such that the @p first value
+   * is the dimension of the highest elongation and the @p second value is the
+   * ratio among the dimensions of the @p cell
+   *
+   * @note This function works in dimension 2 and 3.
+   */
+  template<int dim, int spacedim>
+  std::pair<unsigned int, double>
+  get_longest_direction(typename Triangulation<dim, spacedim>::active_cell_iterator cell);
+
+  /**
+   * Remove hanging nodes from a grid. @p isotropic (default) allows to reproduce
+   * the proportion of each cell in child cells.
+   * Since this procedure is possibly infinite a max number of iteration is
+   * provided
+   *
+   * @param[in,out] tria : tria to refine
+   *
+   * @param[in] isotropic : if true refine cells in each directions, otherwise
+   * (default value) refine the cell in the direction that removes hanging node.
+   *
+   * @param[in] max_iterations : at each step only closest cells to hanging
+   * nodes are refined. In order to remove completely hangind nodes one may have
+   * infinite loops. @p max_iterations is the maximum number of iteration allowed.
+   * If @p max_iterations == 0 this function continues refining until
+   * there are no hanging nodes.
+   */
+  template<int dim, int spacedim>
+  void
+  remove_hanging_nodes( Triangulation<dim,spacedim> &tria,
+                        bool isotropic = false,
+                        unsigned int max_iterations = 100);
```

Comment:
I think this function may also be better suited to the GridTools namespace since it doesn't create a new mesh, just modify an existing one. It may fit into the group of functions such as `GridTools::transform` or `GridTools::distort_random`.